### DISCUSSION TITLE
Kinesis KCL and KPL with Quarkus

### DISCUSSION THREAD

**Discussion by jamesking-github** [quarkusio/quarkus#40738] [Community]
Hi all,

Has anyone looked at, or got experience of, building and running KCL and KPL microservice applications to run in Kubernetes using Quarkus to interact with AWS Kinesis data streams?

I am starting to evaluate Python with Java versus a keep it all in Java set of applications and Quarkus has come up as a solution to the problems that have limited our Java usage recently so any experience or tips that can be shared to lean us one way or another for our use case would be appreciated.

I have already found https://quarkus.io/extensions/io.quarkiverse.amazonservices/quarkus-amazon-kinesis/ however this looks like an alternative to CDK or terraform/terragrunt rather than a basis to build KCL and KPL apps.

Thanks in advance.
James

**Comment by prakashelango** [comment]
recently working with KCL and KPL, https://quarkus.io/extensions/io.quarkiverse.amazonservices/quarkus-amazon-kinesis/ I believe this used aws sdk, when you do use kcl and kpl it's an separate dependency altogether runs as an isolated application.

1. out of box will not get end to end reactive, because kcl uses pull based approach, same kpl used own buffer and timer to send

2. can't produce native builds attempted to build however it failed on kcl and kpl binaries to copy (a good place to explore)

**Reply by jamesking-github** [reply]
Thanks for sharing your insights @prakashelango
